src/Rules/AbstractNodeVisitorRule.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules;
+
+use Twig\Environment;
+use Twig\Node\Node;
+use TwigCsFixer\Report\Report;
+use TwigCsFixer\Report\Violation;
+use TwigCsFixer\Report\ViolationId;
+use Webmozart\Assert\Assert;
+
+abstract class AbstractNodeVisitorRule implements NodeVisitorRuleInterface
+{
+    private ?Report $report = null;
+
+    private ?string $filePath = null;
+
+    /**
+     * @var list<ViolationId>
+     */
+    private array $ignoredViolations = [];
+
+    public function getName(): string
```

Comment:
I would say that `getName`, `getShortName` and `addMessage` can be added in a trait used by both Abstract*Rule. But that could be later.

Reply:
I agree, but let's do that later.